lib/driver-cli.js

```diff
@@ -0,0 +1,249 @@
+import _ from 'lodash';
+import path from 'path';
+import { KNOWN_DRIVERS } from './drivers';
+import DriverConfig, { INSTALL_TYPE_NPM, INSTALL_TYPE_GIT, INSTALL_TYPE_GITHUB,
+                       INSTALL_TYPE_LOCAL } from './driver-config';
+import NPM from './npm';
+import { errAndQuit, log, spinWith, JSON_SPACES } from './cli-utils';
+import { util, fs } from 'appium-support';
+
+async function runDriverCommand (args) {
+  // TODO driver config file should be locked while any of these commands are
+  // running to prevent weird situations
+  let jsonResult = null;
+  const {json, appiumHome} = args;
+  const logFn = (msg) => log(json, msg);
+  const config = new DriverConfig(appiumHome, logFn);
+  const cmd = new DriverCommand({config, json});
+  try {
+    await config.read();
+    jsonResult = await cmd.execute(args);
+  } catch (err) {
+    errAndQuit(json, err);
+  }
+
+  if (json) {
+    console.log(JSON.stringify(jsonResult, null, JSON_SPACES)); // eslint-disable-line no-console
+  }
+}
+
+class DriverCommand {
+
+  constructor ({config, json}) {
+    this.config = config;
+    this.json = json;
+    this.npm = new NPM(this.config.appiumHome);
+  }
+
+  async execute (args) {
+    if (!this[args.driverCommand]) {
```

Comment:
I'd rather use _.isFunction
Also, it makes sense to check directly the instance prototype instead of `this`. We don't want any standard javascript Object methods to match

Reply:
ok makes sense